packages/core/saved-objects/core-saved-objects-migration-server-internal/src/zdt/utils/build_index_mappings.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { cloneDeep } from 'lodash';
+import type { SavedObjectsType } from '@kbn/core-saved-objects-server';
+import {
+  type IndexMapping,
+  type IndexMappingMeta,
+  getModelVersionMapForTypes,
+} from '@kbn/core-saved-objects-base-server-internal';
+import { getBaseMappings, buildTypesMappings } from '../../core';
+
+interface BuildIndexMappingsOpts {
+  types: SavedObjectsType[];
+}
+
+/**
+ * Build the mappings to use when creating a new index.
+ *
+ * @param types The list of all registered SO types.
+ */
+export const buildIndexMappings = ({ types }: BuildIndexMappingsOpts): IndexMapping => {
+  const mappings: IndexMapping = cloneDeep(getBaseMappings());
+  const typeMappings = buildTypesMappings(types);
+
+  mappings.properties = {
+    ...mappings.properties,
+    ...typeMappings,
+  };
+
+  mappings._meta = buildIndexMeta({ types });
+
+  return mappings;
+};
+
+interface BuildIndexMetaOpts {
+  types: SavedObjectsType[];
+}
+
+/**
+ * Build the mapping _meta field to use when creating a new index.
+ *
+ * @param types The list of all registered SO types.
+ */
+export const buildIndexMeta = ({ types }: BuildIndexMetaOpts): IndexMappingMeta => {
+  const modelVersions = getModelVersionMapForTypes(types);
+
+  return {
+    mappingVersions: modelVersions,
+    docVersions: modelVersions,
```

Comment:
so when docVersions has `dashboard: '2'` then the repository will switch to v2 writes? And then once we have transformations we'll have another meta key to indicate that transformation is complete and plugins can switch to v2 queries

Reply:
That's correct